aten/src/TH/generic/THTensorMath.c

```diff
@@ -638,8 +638,40 @@ void THTensor_(scatterAdd)(THTensor *tensor, int dim, THLongTensor *index, THTen
 
   elems_per_row = THLongTensor_size(index, dim);
 
+  // Assumes TENSOR1 is real
+  //         TENSOR2 is src
+  //         TENSOR3 is index
+  // Tests:
+  //   1. index->size[d] <= src->size[d] for all d
+  //   2. index->size[d] <= real->size[d] for all d != dim
+  #define TH_TENSOR_DIM_APPLY3_SIZE_SCATTER(TENSOR1, TENSOR2, TENSOR3, DIMENSION) \
```

Comment:
Because `scatter` and `scatterAdd` both use this, could you please move it out of both functions so that it's written once and not repeated?